R/print.R

```diff
@@ -281,10 +281,209 @@ print.bbi_nonmem_summary <- function(x, .digits = 3, .fixed = FALSE, .off_diag =
   if (!is.null(.nrow)) cat_line(glue("... {orig_rows - .nrow} more rows"), col = "grey")
 }
 
+
+#' @describeIn print_bbi Prints a high level summary of a model from a `bbi_nmboot_summary` object
+#' @importFrom purrr map_chr
+#' @importFrom cli cat_line
+#' @importFrom dplyr mutate_if
+#' @importFrom checkmate assert_number
+#'
+#' @export
+print.bbi_nmboot_summary <- function(x, .digits = 3, .nrow = 10, ...) {
+
+  # print top line info
+  .d <- x[[SUMMARY_DETAILS]]
+  cli_h1("Based on")
+  cat_line(paste("Model:", col_blue(x$based_on_model_path)))
+  cat_line(paste("Dataset:", col_blue(x$based_on_data_set)))
+
+  # Run specifications (seed, stratification columns, cleaned_up)
+  cli_h1("Run Specifications")
+
+  strat_cols <- if(is.null(x$strat_cols)) "None" else paste(x$strat_cols, collapse = ", ")
+  names(strat_cols) <- "Stratification Columns"
+
+  seed <- if(is.null(x$seed)) "None" else x$seed
+  names(seed) <- "Seed"
+
+  run_specs <- c(strat_cols, seed)
+  iwalk(run_specs, ~ cat_bullet(paste0(.y, ": ", col_blue(.x))))
+
+  # Add bullet if cleaned up
+  if(isTRUE(bootstrap_is_cleaned_up(x))){
+    cli::cat_bullet(paste("Cleaned up:", col_green(TRUE)))
+  }
+
+  # Bootstrap run summary (n_samples, any heuristics)
+  cli_h1("Bootstrap Run Summary")
+
+  # TODO: confirm this is appropriate for only_sim (unsure where this comes from)
+  only_sim <- isTRUE("only_sim" %in% names(.d))
+  if (only_sim) {
+    cat_line("No Estimation Methods (ONLYSIM)\n")
+  } else {
+    cli::cat_line("Estimation Method(s):\n")
+    purrr::walk(paste(x$estimation_method, "\n"), cli::cat_bullet, bullet = "en_dash")
+  }
```

Comment:
@seth127 this is one section I wasnt sure of and didn't _really_ attempt to address. The rest of the `TODO`s I left for now (about exposing `resample_df` arguments to users relating to ID/Key columns, your `.overwrite` argument `TODO`, and an existing one from outside of this PR):
![image](https://github.com/metrumresearchgroup/bbr/assets/18685116/a0c2bb3e-92f6-4984-955f-1e124523845a)

Reply:
Hmm, I'm not sure about that either. I assume it comes from copying [this section from `bbi_nonmem_summary`](https://github.com/metrumresearchgroup/bbr/blob/e502d1a44fe4d326909fa9fafebfdde8bde5c034/R/print.R#L207-L214) above.

That section was added in https://github.com/metrumresearchgroup/bbr/commit/bbe68ed0fbb3a5bb299cf769d795c8208b1c2226 and the commit message explains the rationale pretty well.

In terms of what to do here... You may want to check, but I don't think it's possible to have gotten to this point (i.e. a `bbi_nmboot_summary` object that would trigger this print method) with an `ONLYSIM` model. I feel like it would fail somewhere earlier in the bootstrap pipeline. It's probably worth testing what happens when you feed a model with `$SIM` and no `$EST` to `new_bootstrap_model()`...

If I'm right though, you can probably delete this check here.

Related: you probably want to revisit those [those `bbi_nonmem_summary` lines](https://github.com/metrumresearchgroup/bbr/blob/e502d1a44fe4d326909fa9fafebfdde8bde5c034/R/print.R#L207-L214) as part of #687